responsive-test-utils/src/test/java/dev/responsive/kafka/api/ResponsiveTopologyTestDriverKeyValueStoreTest.java

```diff
@@ -155,56 +169,104 @@ public void shouldEnforceKeyBasedTtlByAdvancingWallclockTime(final KVSchema type
 
-    bids.pipeInput("c", "c,102,1", 20);       // time = 20 -- no result b/c alice has expired
-    bids.pipeInput("d", "d,103,3", 20);       // time = 20 -- result since carol is not expired
+      bids.pipeInput("c", "c,102,1", 20);       // time = 20 -- no result b/c alice has expired
+      bids.pipeInput("d", "d,103,3", 20);       // time = 20 -- result since carol is not expired
 
-    people.pipeInput("1", "1,alex,CA", 20);   // insert time = 20
-    bids.pipeInput("e", "e,104,1", 20);       // time = 20 -- result as alex has replaced alice
+      people.pipeInput("1", "1,alex,CA", 20);   // insert time = 20
+      bids.pipeInput("e", "e,104,1", 20);       // time = 20 -- result as alex has replaced alice
 
-    driver.advanceWallClockTime(Duration.ofMillis(30)); // advances wallclock time to 30
+      driver.advanceWallClockTime(Duration.ofMillis(30)); // advances wallclock time to 30
 
-    bids.pipeInput("f", "f,105,3", 30);      // time = 30 -- no result b/c carol has expired
+      bids.pipeInput("f", "f,105,3", 30);      // time = 30 -- no result b/c carol has expired
 
-    bids.pipeInput("g", "g,106,0", 30);      // time = 30 -- result b/c person w/ id 0 is infinite
+      bids.pipeInput("g", "g,106,0", 30);      // time = 30 -- result b/c person w/ id 0 is infinite
 
-    // Then:
-    final List<String> outputs = output.readValuesToList();
-    MatcherAssert.assertThat(outputs, Matchers.contains(
-        "a,100,1,1,alice,CA",
-        "d,103,3,3,carol,CA",
-        "e,104,1,1,alex,CA",
-        "g,106,0,0,infinite,CA"
-    ));
-    driver.close();
+      // Then:
+      final List<String> outputs = output.readValuesToList();
+      MatcherAssert.assertThat(outputs, Matchers.contains(
+          "a,100,1,1,alice,CA",
+          "d,103,3,3,carol,CA",
+          "e,104,1,1,alex,CA",
+          "g,106,0,0,infinite,CA"
+      ));
+    }
+  }
+
+  @ParameterizedTest
+  @EnumSource(SchemaTypes.KVSchema.class)
+  public void shouldDeduplicateWithTtlProviderToExpireOldRecords(final KVSchema type) {
+    final String value = "ignored";
+
+    final String key = "key";
+    final long ttlMs = 10_000L; // 10s
+    final var ttlProvider =
+        TtlProvider.withDefault(Duration.ofMillis(ttlMs))
+            .fromKey(
+                k -> {
+                  final Duration ttl = Duration.ofMillis(ttlMs);
+                  System.out.println("TTL: " + ttl);
+                  return Optional.of(TtlDuration.of(ttl));
+                });
+
+    final ResponsiveKeyValueParams params = paramsForType(type).withTtlProvider(ttlProvider);
+    final Topology topology = deduplicatorApp("input", "output", params);
+    try (final ResponsiveTopologyTestDriver driver = setupDriver(topology)) {
+
+      final TestInputTopic<String, String> inputTopic = driver.createInputTopic(
+          "input",
+          new StringSerializer(),
+          new StringSerializer(),
+          STARTING_TIME,
```

Comment:
Once we fix this issue in Streams we should be able to remove this explicit time initialization here and in the driver itself, and have the tests still pass. Until then, we can't verify ttl with millisecond precision because the system time might drift between the driver initialization and the test topic creation